orangecontrib/spectroscopy/widgets/owpeakfit.py

```diff
@@ -349,28 +376,45 @@ def progress_interrupt(i: float):
         # happen when adding a preprocessor (there, commit() is called twice).
         # Wait 100 ms before processing - if a new task is started in meanwhile,
         # allow that is easily` cancelled.
-        for i in range(10):
-            time.sleep(0.005)
+        for _ in range(10):
+            time.sleep(0.010)
             progress_interrupt(0)
 
         model, parameters = create_composite_model(m_def)
 
         data_fits = data_anno = data_resid = None
         if data is not None and model is not None:
             orig_data = data
-            output = init_output_array(data, model, parameters)
+            output = []
             x = getx(data)
             n = len(data)
             fits = []
             residuals = []
-            for row in data:
-                i = row.row_index
-                out = model.fit(row.x, parameters, x=x)
-                add_result_to_output_array(output, i, out, x)
-                fits.append(out.eval(x=x))
-                residuals.append(out.residual)
-                progress_interrupt(i / n * 100)
-            data = fit_results_table(output, out, orig_data)
+
+            with multiprocessing.Pool(processes=None,
+                                      initializer=pool_initializer,
+                                      initargs=(model.dumps(), parameters, x)) as p:
+                res = p.map_async(pool_fit, data.X, chunksize=1)
```

Comment:
I had previously seen that chunksize needed to be optimized, so I tested it (N=736, collagen.csv, quad-core CPU):
````
with chunksize 1: 17.978233337402344
with chunksize 10: 18.698899269104004
with chunksize 100: 17.8134982585907
with chunksize 1000: 48.91136884689331
````

To my surprise this implementation has no measureable start-up overhead on my machine! Nice.

Reply:
The start-up overhead is certainly there, but is equal in all your tests. :)

I think Chunking could only improve performance for data sets with a very low number of wavenumbers, which are fitted more easily. If we went with it, I am for making it dependant on the length of x.

Reply:
I see. Let's leave it.